cmake/FindRust.cmake

```diff
@@ -221,6 +226,24 @@ directory = \".cargo/vendor\"
     endif()
 endfunction()
 
+# Export the project name (including upstream project) and maintainer mode
+# status for use by build scripts
+function(GetCargoExtraEnv result)
+    set(_result "PROJECT_NAME=${PROJECT_NAME}")
+    list(APPEND _result "CMAKE_PROJECT_NAME=${CMAKE_PROJECT_NAME}")
+    list(APPEND _result "MAINTAINER_MODE=${MAINTAINER_MODE}")
+    # Propagate any additional Rust compiler flags
+    list(APPEND _result "RUSTFLAGS=${RUSTFLAGS}")
+    set("${result}" "${_result}" PARENT_SCOPE)
+endfunction()
+
+function(GetCargoLocalDirEnv result)
```

Comment:
Three things:
1. Naming convention for CMake functions seems to be snake_case. Can we switch this from CamelCase?

2. I wonder if this should be defined before the `cargo_vendor` function as it is first used there. 

3. This function could use a comment block

Reply:
1. Changed.
2. Doesn't seem to matter thanks to lazy evaluation, but there could be more tidying done
3. Added.